packages/medusa/src/loaders/config.ts

```diff
@@ -9,18 +9,35 @@ const errorHandler = isProduction
     }
   : console.log
 
-export default (rootDirectory: string): ConfigModule => {
-  const { configModule } = getConfigFile(rootDirectory, `medusa-config`) as {
-    configModule: ConfigModule
+export default async (rootDirectory: string): Promise<ConfigModule> => {
+  /** we promisify the configuration to support both synchronouse and asynchronous loading.  */
+
+  const configuration = getConfigFile(rootDirectory, `medusa-config`)
+  const resolveConfigProperties = async (obj): Promise<ConfigModule> => {
+    for (const key of Object.keys(obj)) {
+      if (typeof obj[key] === "object" && obj[key] !== null) {
+        await resolveConfigProperties(obj[key])
+      }
+      if (typeof obj[key] === "function") {
+        obj[key] = await obj[key]()
+      }
+    }
+    return obj
   }
+  let configModule = await resolveConfigProperties(configuration)
```

Comment:
todo: this can be a const, you should get the config object as the return

Reply:
done